### PR TITLE
Web Inspector: Unify font sizes in Audits

### PR DESCRIPTION
**Pull request by francescorn**
#### 743746ac72bddd0053b72713f839c635d0e86ee6
<pre>
Web Inspector: Unify font sizes in Audits
<a href="https://bugs.webkit.org/show_bug.cgi?id=224128">https://bugs.webkit.org/show_bug.cgi?id=224128</a>

Reviewed by Devin Rousso.

Change the header in audit tab from 2.8em to 2em.

Amend changes:
* Source/WebInspectorUI/UserInterface/Views/AuditTestContentView.css:
(.content-view.audit-test &gt; header p):

Combined changes:
* Source/WebInspectorUI/UserInterface/Views/AuditTestContentView.css:
(.content-view-container &gt; .content-view.audit-test &gt; header h1):

Canonical link: <a href="https://commits.webkit.org/273615@main">https://commits.webkit.org/273615@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/dc69911bb57f6fc17f482ecf9b5b12d14be7f316

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [⏳ 🧪 style ](https://ews-build.webkit.org/#/builders/Style-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/WinCairo-EWS "Waiting in queue, processing has not started yet") 
| [⏳ 🧪 bindings ](https://ews-build.webkit.org/#/builders/Bindings-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitperl ](https://ews-build.webkit.org/#/builders/WebKitPerl-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitpy ](https://ews-build.webkit.org/#/builders/WebKitPy-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk1 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 gtk-wk2 ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-arm64-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 services ](https://ews-build.webkit.org/#/builders/Services-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2-stress ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8202 "Built successfully and passed tests") | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-10-Build-EWS "Waiting in queue, processing has not started yet") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-10-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | |

### CODE REVIEW

File: Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css [hunk 1]
```diff
@@ -78,6 +78,10 @@
     margin-inline-end: 0.25em;
 }
 
+.content-view.audit-test-case > header h1 > span {
```
Comment (dcrousso) [inline Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css]: are you trying to adjust the font size of only the test case header, or all audit headers (i.e. including groups)?  if the latter, i think you might want to adjust `.content-view-container > .content-view.audit-test > header h1` instead

**Review by dcrousso — COMMENTED** [review]
@francescorn  can you attach a screenshot of what the Audit Tab looks like after this change?

it was part of the original design to have these differently sized font headers, so im hesitant to adjust anything without making sure everything still works together

File: Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css (hunk 1 above)
Comment (francescorn) [reply Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css:81]: It looks like the header should be modified according to the bug, so I adjusted the header to be smaller.

Comment (francescorn) [reply Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css]: Only the test case header. I adjusted `.content-view-container > .content-view.audit-test > header h1` from 2.8em to 1.5em with your suggestion.

**Comment by francescorn** [comment]
Here is the mini-browser comparing the minibrowser on the left with my change to what is currently implemented in Safari on the right in the audit tab. 
audit-tab-adjusted-font

**Comment by francescorn** [comment]
I agree that the header looks a little big in the original audit tab.

**Comment by francescorn** [comment]
Why was it part of the original design to have the font headers sized differently? @dcrousso

**Comment by dcrousso** [comment]
> I am comparing the mini-browser on the left with my change to what is currently implemented in Safari on the right in the audit tab. audit_font_tab

can you show what it looks like when an audit group is selected?  i want to make sure that the "cascade" of header font sizes still looks good

> Why was it part of the original design to have the font headers sized differently? @dcrousso

you'd have to ask the original designer 😅

i think the intention was that this data is much more presentational than almost everything else in Web Inspector (e.g. it's more about "how many tests passed?" and whatnot than "please show me the source code")

**Comment by francescorn** [comment]
I am closing the pr, see pr #22633. https://github.com/WebKit/WebKit/pull/22633#issuecomment-1888462325

**Comment by francescorn** [comment]
Reopening this pr because pr #22633 is not necessary.

**Comment by francescorn** [comment]
Can we merge this pr? @dcrousso @rcaliman-apple @cookiecrook @NV

**Comment by francescorn** [comment]
Screenshot 2024-01-19 at 10 55 38 AM @dcrousso

**Comment by dcrousso** [comment]
it seems like the `font-size` of the title of both the parent group and child group(s) are now the same (e.g. "Demo Audit" and "Result Levels" look very similarly sized)

im not sure that's ideal as it removes an aspect of visual hierarchy that's arguably necessary (i.e. is "Demo Audit" the name of the (currently selected) top-level group or is it the name of an individual test within some other top-level group that for some reason has no output/content?)

i think we need to keep some sort of distinction (and im not necessarily saying it has to be `font-size`, just that there needs to be something ideally obvious) for "this is the currently selected top-level group" as otherwise it'll look too similar to any subgroups

**Comment by francescorn** [comment]
Should I create an h2 class? Both of the headers are h1. 
inspecting_the_audit_tab_both_headers_in_demo_audit

**Comment by francescorn** [comment]
Here are my results from the patch with the mini-browser with my changes on the right and Safari on the left: section h1 with a font size of 1.5em, header p with a font size of 1.35em, and header h1 with a font size of 1.5em, and created .information > p with a font size of 1.1em.
minibrowser_safari_audit_tab_font_changes

**Comment by francescorn** [comment]
The mini-browser on the left with my changes vs Safari on the right. audit_tab

**Comment by francescorn** [comment]
I am changing the font-size of content-view-container > .content-view.audit-test > header h1 only, from 2.8em to 2em in order to address the bug. The minibrowser is on the right with my change, and Safari is on the left.
audit_tab_header_changed

**Review by dcrousso — APPROVED** [review]
nice change!  i agree this looks better :)

thanks for iterating on this with me!

**Comment by francescorn** [comment]
Thank you @dcrousso

**Pull request merged**